### PR TITLE
[feature] throttle max requests to avoid rate limits

### PR DESCRIPTION
**Pull request by icereval**

### CODE REVIEW

File: waterbutler/providers/googledrive/provider.py [hunk 1]
```diff
@@ -288,6 +288,10 @@ def create_folder(self, path, **kwargs):
 
         return GoogleDriveFolderMetadata((yield from resp.json()), path)
 
+    @asyncio.coroutine
+    def make_request(self, *args, **kwargs):
+        return (yield from super().make_request(*args, **kwargs))
+
```
Comment (felliott) [inline waterbutler/providers/googledrive/provider.py]: I might be missing something here, but isn't this redundant?

File: waterbutler/tasks/settings.py [hunk 2]
```diff
@@ -42,5 +42,6 @@
     'waterbutler.tasks.move'
 ])
 
+CELERY_ACKS_LATE = True
```
Comment (felliott) [inline waterbutler/tasks/settings.py]: This doesn't seem to be used anymore.

Comment (icereval) [inline waterbutler/tasks/settings.py]: Looks like we need to get this turned on in more places http://docs.celeryproject.org/en/latest/faq.html#should-i-use-retry-or-acks-late

File: waterbutler/providers/googledrive/provider.py (hunk 1 above)
Comment (chrisseto) [inline waterbutler/providers/googledrive/provider.py]: 100% redundant, nice catch!
I think I had originally throttled per provider but later moved the throttle to be global again.

File: waterbutler/tasks/settings.py (hunk 2 above)
Comment (felliott) [inline waterbutler/tasks/settings.py]: Derp.  That was me being dumb, thinking it was an unused local config setting.

**Pull request merged**